rest/rest-smartproxy/src/main/java/org/ow2/proactive_grid_cloud_portal/smartproxy/RestSmartProxyImpl.java

```diff
@@ -363,14 +376,22 @@ protected void downloadTaskOutputFiles(AwaitedJob awaitedjob, String jobId, Stri
             return;
         }
 
-        String outputSpace = awaitedjob.getOutputSpaceURL();
         String sourceFile;
         try {
-            String userSpace = getLocalUserSpace();
-            if (!outputSpace.startsWith(userSpace)) {
-                logger.warn("RestSmartProxy does not support data transfers outside USERSPACE.");
+            List<String> userSpaces = getUserSpaceURIs();
+            List<String> outputSpaces = Arrays.asList(Tools.dataSpaceConfigPropertyToUrls(awaitedjob.getOutputSpaceURL()));
+
+            sourceFile = extractRemotePath(userSpaces, outputSpaces);
+
+            if (sourceFile == null) {
+                if (awaitedjob.isAutomaticTransfer()) {
+                    logger.error("Could not extract remote path using inputSpace=" + outputSpaces + " and userSpace=" +
+                                 userSpaces);
+                } else {
+                    throw new IllegalArgumentException("Could not extract remote path using inputSpace=" +
+                                                       outputSpaces + " and userSpace=" + userSpaces);
+                }
             }
-            sourceFile = outputSpace.substring(userSpace.length() + 1);
         } catch (Throwable error) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)